app/modules/tunnel/socket_service.py

```
@@ -0,0 +1,231 @@
+        return sid
+
+    def is_workspace_online(self, workspace_id: str) -> bool:
+        """Synchronous check: is there a socket registered for this workspace_id?"""
+        if not self._redis_url:
+            return False
+        try:
+            import redis
+            sync_redis = redis.from_url(self._redis_url, decode_responses=True)
+            key = _workspace_socket_key(workspace_id)
+            return sync_redis.exists(key) > 0
+        except Exception:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)